woocommerce/woocommerce
# Create Extension: Add dynamic config and Woo version number

### All Submissions:






### Changes proposed in this Pull Request:



This PR wraps the `module.exports` contents in a self invoked async function so that we can `await` any API calls with relevant data. In this case, we gather the latest version of WooCommerce from the wordpress.org API so that wp-env can be correctly configured with the latest version.

Closes https://github.com/woocommerce/platform-private/issues/130






### How to test the changes in this Pull Request:



Testing this PR requires a development version of Gutenberg.

1. Pull down Gutenberg and checkout `add/create-block-dynamic-config` from this PR https://github.com/WordPress/gutenberg/pull/46456
2. `npm install` and make sure you are on Node v14 for this step.
3. In this repo, be sure to `pnpm install` as I've added dependencies.
4. Run this command from a new folder.
```
npx ../path/to/gutenberg/packages/create-block -t ../path/to/woocommerce/packages/js/create-extension my-extension-name
```
5. Check the resulting project, it should have a `.wp-env.json` file correctly configured with a download link to the latest WooCommerce.



### Other information:




### FOR PR REVIEWER ONLY:
